### ISSUE TITLE
[Release 1.31] Bump crictl

### ISSUE THREAD

**Issue by cwayne18** [rancher/rke2#6976]
Bump to 1.31.1-build20241008

**Comment by aganesh-suse** [comment]
Issue: Getting crictl version unknown. 
Expected result: Display the correct crictl version

OS: Ubuntu 24.04 

Branch: release-1.31
Commit ID: a70094f7f64b95192f7bf0b4a282c4007b6c8b12

```
$ rke2 -v
rke2 version v1.31.1+dev.a70094f7 (a70094f7f64b95192f7bf0b4a282c4007b6c8b12)
go version go1.22.6 X:boringcrypto
```

```
$ crictl -v
crictl version unknown
```

**Comment by brandond** [comment]
I suspect https://github.com/rancher/image-build-crictl/pull/40 made the repo dirty and broke whatever logic the upstream build script was using to set the version. I can fix that and rebuild/update.

**Comment by brandond** [comment]
The package has changed for cri-tools v1.31: https://github.com/kubernetes-sigs/cri-tools/commit/3f6ce817f73e5eda8ff12c832bf3ce19fed02659

**Comment by ShylajaDevadiga** [comment]
### Validated fix the latest commit on release-1.31 branch.

```
ubuntu@ip-172-31-1-149:~$ rke2 -v
rke2 version v1.31.1+dev.cc064cab (cc064cab2d601a777e0a75bc2602068f799fceb8)
go version go1.22.6 X:boringcrypto
```
```
ubuntu@ip-172-31-1-149:~$ crictl -v
crictl version v1.31.1
```
